doc/concepts/configuration.rst

```diff
@@ -0,0 +1,618 @@
+
+..  literalinclude:: /code_snippets/test/config/iproto_instance_scope.yaml
+    :language: yaml
+    :dedent:
+
+-   The ``instances`` section includes only one instance named *instance-001*.
+    The ``iproto.listen`` option sets a port used to listen for incoming requests.
+-   The ``replicasets`` section contains one replica set named *replicaset-001*.
+-   The ``groups`` section contains one group named *group-001*.
+
+
+.. _configuration_scopes:
+
+Configuration scopes
+********************
+
+This section shows how to control a scope the specified configuration option is applied to.
+Most of the configuration options can be applied to a specific instance, replica set, group, or to all instances globally.
+
+-   *Instance*
+
+    To apply specific configuration options to a concrete instance,
+    specify such options for this instance only.
+    In the example below, ``iproto.listen`` is applied to *instance-001* only.
+
+    ..  literalinclude:: /code_snippets/test/config/iproto_instance_scope.yaml
+        :language: yaml
+        :emphasize-lines: 6-8
+        :dedent:
+
+-   *Replica set*
+
+    In this example, ``iproto.listen`` is in effect for all instances in *replicaset-001*.
+
+    ..  literalinclude:: /code_snippets/test/config/iproto_replicaset_scope.yaml
+        :language: yaml
+        :emphasize-lines: 4-6
+        :dedent:
+
+-   *Group*
+
+    In this example, ``iproto.listen`` is in effect for all instances in *group-001*.
+
+    ..  literalinclude:: /code_snippets/test/config/iproto_group_scope.yaml
+        :language: yaml
+        :emphasize-lines: 2-4
+        :dedent:
+
+-   *Global*
+
+    In this example, ``iproto.listen`` is applied to all instances of all groups.
+
+    ..  literalinclude:: /code_snippets/test/config/iproto_global_scope.yaml
+        :language: yaml
+        :emphasize-lines: 1-2
+        :dedent:
+
+
+.. NOTE::
+
+    The :ref:`Configuration reference <configuration_reference>` contains information to which scopes each configuration option can be applied.
+
+
+.. _configuration_replica_set_scopes:
+
+Replica set configuration and configuration scopes
+**************************************************
+
+This section shows how to configure a :ref:`replica set <replication>` with a manual failover
+and describes how specific configuration options work in different configuration scopes.
+
+..  literalinclude:: /code_snippets/test/config/replicaset_manual.yaml
+    :language: yaml
+    :dedent:
+
+-   ``credentials`` (*global*)
+
+    Options in this section grant the specified privileges to the *replicator* user used for replication and
+    the *client* user that can perform any action.
+    These options are applied globally to all instances.
+
+-   ``iproto`` (*global*, *instance*)
+
+    The ``iproto`` section is specified on both global and instance levels.
+    The ``iproto.advertise.peer`` option specifies a user name and a host used by replicas to connect to each other.
```

Comment:
Is it describe an idea of `iproto.advertise.peer` at whole or in regards of this particular example. It seems, it is inaccurate in both senses.

There are four variants of `iproto.advertise.peer` syntax, described in `instance_config.lua`.

In this particular example `username@` syntax is used. The option has no host information.

Reply:
It seems that PR #3825 fixes the inaccuracy.